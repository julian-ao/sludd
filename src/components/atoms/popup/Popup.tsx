import './popup.css';

export type PopupProps = {
    text: string;
    show: boolean;
};

const Popup = (props: PopupProps) => {
    return (
        <div className={`popup ${props.show ? 'show' : ''}`}>
            <div className="popup-content">
                <p>{props.text}</p>
            </div>
        </div>
    );
};

export default Popup;
