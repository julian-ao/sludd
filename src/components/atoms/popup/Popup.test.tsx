import { cleanup, render } from '@testing-library/react';
import { afterEach } from 'vitest';
import Popup from './Popup';

afterEach(() => {
    cleanup();
});

describe('Popup', () => {
    it('should render correctly', () => {
        const { container } = render(<Popup
            text="test"
            show={true}
        />);
        expect(container).toMatchSnapshot();
    });
});
