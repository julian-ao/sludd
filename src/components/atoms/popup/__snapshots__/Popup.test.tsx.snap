// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Popup > should render correctly 1`] = `
<div>
  <div
    class="popup show"
  >
    <div
      class="popup-content"
    >
      <p>
        test
      </p>
    </div>
  </div>
</div>
`;
